src/components/annotation-widgets/default/MainImage.vue

```diff
@@ -68,25 +82,29 @@ export default class MainImage extends Vue {
     @Prop({required: true, type: String})
     colormapName!: string
     @Prop({required: true, type: Number})
+    zoom!: number
+    @Prop({required: true, type: Number})
     opacity!: number
     @Prop({required: true})
     imageLoaderSettings!: any
     @Prop({required: true, type: Function})
     onImageMove!: Function
-    @Prop({required: true, type: Function})
-    onImageZoom!: Function
+
+    get zoomVal(): number {
+      return this.zoom
+    }
 
     get colorbarDirection(): string {
       return this.colormap[0] == '-' ? 'bottom' : 'top';
     }
 
     saveImage(event: any): void {
-      let node = this.$refs.imageLoader.getContainer();
+      let node = this.$refs.imageLoader.$refs.parent
```

Comment:
This is the integration point between two components, which makes it very fragile, and care should be taken not only to not break it, but also to leave it in a state where it's not easy to accidentally break in future changes. 

This change creates the first place in the codebase where a component directly accesses the `$refs` of a *different* component - in all other places we've put `$refs` access behind a getter like what was here previously with `getContainer`. As a result of this pattern, it's generally pretty obvious which methods on a component are potentially called by the parent component. I'd prefer if this code was changed back to calling a function on `$refs.imageLoader` to keep with the existing pattern. 

If `ImageLoader.vue` was a vue class component, we could actually get TypeScript to enforce safety here by giving `$refs` a type e.g. `interface MainImageRefs { imageLoader: ImageLoader }`, and in that case it wouldn't matter as breakages would be quickly detected. Unfortunately that's not the case yet, so we should stick to established patterns to avoid unintentional breakages in a refactor.

Reply:
Good point, the only problem I've discovered was that when one wants to save his image which was acquired by getContainer() was that it will be a predefined aspect ratio(since ` let node = this.$refs.imageLoader.getContainer();` will be pointing to the div element that doesn't change with scaling, even if the image was zoomed.

Reply:
You could create a new function `getParent()` on ImageLoader.

I also noticed that `getContainer` isn't used anywhere else, so you could also just change it to return a different element.